lib/perl/Genome/Model/Tools/Validation/ProcessSomaticValidation.pm

```diff
@@ -751,30 +705,86 @@ sub execute {
       `grep -w Somatic $small_indel_file | cut -f 1-5 >$output_dir/$sample_name/indels/small.indels`;
       `cat $output_dir/$sample_name/indels/small.indels $output_dir/$sample_name/indels/large.indels | joinx sort >$output_dir/$sample_name/indels/indels.hq`;
   }
+  my $indel_file = "$output_dir/$sample_name/indels/indels.hq";
+  
 
-
-
+  #SVs
+  my $process_svs = $self->process_svs;
+  my $sv_file = "$build_dir/validation/sv/assembly_output.csv.merged.readcounts.somatic.wgs_readcounts.somatic";
+  if($process_svs){
+      unless( -e $sv_file ){
+          $self->warning_message("SV results for $sample_name not found, skipping SVs");
+          $process_svs = 0;
+      }
+  }
   if($process_svs){
       `mkdir $output_dir/$sample_name/svs`;
       `cp $sv_file $output_dir/$sample_name/svs/svs.hq` unless( -e "$output_dir/$sample_name/svs/$sv_file");
   }
 
-  #only need to operate on the non-zero files
-  @snv_files = ();
-  push(@snv_files,"$output_dir/$sample_name/snvs/snvs.validated") if (-s "$output_dir/$sample_name/snvs/snvs.validated");
-  push(@snv_files,"$output_dir/$sample_name/snvs/snvs.notvalidated") if (-s "$output_dir/$sample_name/snvs/snvs.notvalidated");
-  push(@snv_files, "$output_dir/$sample_name/snvs/" . basename($newcalls)) if (-s "$output_dir/$sample_name/snvs/" . basename($newcalls));
-
-  my $indel_file = "$output_dir/$sample_name/indels/indels.hq";
-
 
+  
   #clean the files
   my $i=0;
   for($i=0;$i<@snv_files;$i++){
       $snv_files[$i] = cleanFile($snv_files[$i])
   }
   $indel_file = cleanFile($indel_file);
 
+  #-------------------------------------------------
+  #filter out the off-target regions, if target regions are available
+  if($self->restrict_to_target_regions){
+      $self->status_message("Filtering out off-target regions...");
+      
+      my $featurelist;
+      if($self->target_regions) {
+          $featurelist = $self->target_regions;
+      } else { 
+          $featurelist = $model->target_region_set->file_path;
+      }
+
+
+      if(defined($featurelist) && (-s $featurelist)){
+          #clean up feature list
+          open(FEATFILE,">$output_dir/$sample_name/featurelist.tmp");
+          my $inFh = IO::File->new( $featurelist ) || die "can't open file feature file\n";
+          while( my $line = $inFh->getline )
+          {
+              chomp($line);
+              next if $line =~ /^track/;
+              my ( $chr, $start, $stop, @rest) = split( /\t/, $line );
+              #remove chr if present
+              $chr =~ s/^chr//g;
+              print FEATFILE join("\t",( $chr, $start, $stop, @rest)) . "\n";
+          }
+          close($inFh);
+          close(FEATFILE);
+          my $cmd;
+          $cmd = "joinx sort '" . $output_dir . "/" . $sample_name . "/featurelist.tmp'" . " >'" . $output_dir . "/" . $sample_name . "/featurelist'";
+          `$cmd`;
+          $cmd = "rm -f '" . $output_dir . "/" . $sample_name . "/featurelist.tmp'";
+          `$cmd`;
```

Comment:
If `Genome::Sys->shellcmd` is overkill, I'd still recommend using `system` with the params in an `@array`).  Maybe even better would be to `Genome::Utility::Text::sanitize_name_for_filesystem` the `$sample_name`!